### PR TITLE
refactor!: igraph_get_biadjacency() now takes a weights parameter

### PR DESCRIPTION
**Pull request by szhorvat**
This completes the breaking portion of #2414 so we can make progress towards 1.0.

### CODE REVIEW

**Comment by szhorvat** [comment]
@ntamas Can you have a look? Let me know what you think about #2579, and let's do that change here as well.

File: src/misc/bipartite.c [hunk 1]
```diff
@@ -625,48 +625,49 @@ igraph_error_t igraph_incidence(
  * \param types Pointer to an initialized boolean vector, or a null
  *   pointer. If not a null pointer, then the vertex types are stored
  *   here. It is resized as needed.
- * \param input The bipartite adjacency matrix that serves as an input
+ * \param biadjmatrix The bipartite adjacency matrix that serves as an biadjmatrix
```
Comment (ntamas) [inline src/misc/bipartite.c:628]: `an biadjmatrix` is grammatically incorrect

File: src/misc/bipartite.c [hunk 2]
```diff
@@ -625,48 +625,49 @@ igraph_error_t igraph_incidence(
...
-    }
-
     if (multiple) {
 
-        for (i = 0; i < n1; i++) {
-            for (j = 0; j < n2; j++) {
-                igraph_integer_t elem = ceil(MATRIX(*input, i, j));
+        if (n1 > 0 && n2 > 0 && igraph_matrix_min(biadjmatrix) < 0) {
+            IGRAPH_ERRORF(
+                "Bipartite adjacencey matrix elements should be non-negative, found %g.",
```
Comment (ntamas) [inline src/misc/bipartite.c:663]: typo here; should be `adjacency`

**Review by ntamas — CHANGES_REQUESTED** [review]
Looks good to me in general; I only proposed fixes for a few typos

**Review by ntamas — APPROVED** [review]
LGTM.

**Pull request merged**